### ISSUE TITLE
Should Pure derivation utilize Empty for constants (e.g. primitive values)?

### ISSUE THREAD

**Issue by joroKr21** [typelevel/kittens#125]
This came up in #104 but it deserves its own issue.

For example we could derive `Pure[Interleaved]` where:
```scala
case class Interleaved[T](i: Int, t: T, d: Double, tt: List[T], s: String)
```

if we use the `Empty[Int]`, `Empty[Double]` and `Empty[String]` instances.

**Comment by milessabin** [comment]
Yes, I think that would be reasonable. Or a `Monoid`. The key thing is that you have to be able to construct a value given a single `T`.

**Comment by joroKr21** [comment]
Monoid should give us Empty I think.

**Comment by milessabin** [comment]
Do you mean in the sense of,
```scala
implicit def monoidIsEmpty[T](implicit mt: Monoid[T]): Empty[T] = ...
```
In which case, yes :-)

**Comment by joroKr21** [comment]
Fixed by #151
